distributednetworkmanager/src/main/java/com/gruppo1/distributednetworkmanager/KadAction.java

```diff
@@ -0,0 +1,348 @@
+            } catch (NumberFormatException e) {
+                throw new IllegalArgumentException(e.getMessage() + "\n" + CURRENT_MESSAGE_NOT_FOUND_ERROR_MSG);
+            }
+            try {
+                payloadType = PayloadType.getTypeFromCode(Integer.parseInt(parameteres[PAYLOAD_TYPE_POSITION]));
+            } catch (NumberFormatException e) {
+                throw new IllegalArgumentException(e.getMessage() + "\n" + PAYLOAD_TYPE_NOT_FOUND_ERROR_MSG);
+            }
+            payload = parameteres[PAYLOAD_POSITION];
+            if (!isValid())
+                throw new IllegalArgumentException();
+        }
+    }
+
+    /**
+     * Check if all the action's parameters are valid
+     *
+     * @return True if the defined action is valid and fits into a Message.
+     */
+    @Override
+    public boolean isValid() {
+        if (this.actionType.equals(ActionType.INVALID))
+            return false;
+        if (this.payloadType.equals(PayloadType.INVALID))
+            return false;
+        if (currentPart <= 0 || totalParts <= 0 || currentPart > totalParts)
+            return false;
+        if (!payloadMatchesType(payloadType, payload))
+            return false;
+        try {
+            toMessage();
+        } catch (InvalidMessageException e) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Check if payload's content type matches with payloadType passed as parameter.
+     *
+     * @param payloadType The type of the payload.
+     * @param payload The string which contains the value of the payload.
+     * @return True if the payload string contains a value of the correct type,false otherwise.
+     */
+    public static boolean payloadMatchesType(PayloadType payloadType, String payload) {
+        switch (payloadType) {
+            case IGNORED:
+            case BOOLEAN:
+                return true;
+            case NODE_ID:
+                BitSet bitset=BitSetUtils.decodeHexString(payload);
+                return bitset.length()==ID_LENGTH;
+            case PEER_ADDRESS:
+                return SMSPeer.isAddressValid(payload) == SMSPeer.PhoneNumberValidity.ADDRESS_VALID;
+            case RESOURCE:
+                try {
+                    return StringResource.parseString(payload, SEPARATOR).isValid();
+                }
+                catch (IllegalArgumentException e){
+                    return false;
+                }
+            default:
+                return false;
+        }
+
+    }
+
+    /**
+     * Get a message with correct formatted action command ready to be sent
+     *
+     * @return A Message containing the formatted action command ready to be sent.
+     */
+    @Override
+    public SMSMessage toMessage() {
+        StringBuilder body = new StringBuilder();
+        body.append(actionType.getCode()).append(SEPARATOR).append(operationId)
+                .append(SEPARATOR).append(currentPart).append(SEPARATOR).append(totalParts)
```

Comment:
Here currentPart, totalParts and operationId should be checked for padding zeros if we're using the option I suggested above